src/main/java/org/kitteh/irc/client/library/util/IPUtil.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * * Copyright (C) 2013-2016 Matt Baxter http://kitteh.org
+ *
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use, copy,
+ * modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.kitteh.irc.client.library.util;
+
+import java.math.BigInteger;
+import java.net.Inet6Address;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Arrays;
+
+public final class IPUtil {
+    private static final BigInteger X_FFFF = BigInteger.valueOf(0xFFFF);
+    // Length of (2**128) - 1 -- IPv6 has up to 2**128 addresses
+    private static final int LENGTH_OF_LARGEST_IPV6_INTEGER = 39;
+
+    private IPUtil() {
+    }
+
+    /**
+     * Takes a string containing an integer (potentially larger than 32-bits) and converts it to the appropriate IP address class.
+     *
+     * @param ipInteger the IP integer string
+     * @return the corresponding IP address as an InetAddress instance
+     */
+    public static InetAddress getInetAdressFromIntString(String ipInteger) {
```

Comment:
Missing a `d` in `address`.
`getInetAdressFromIntString`
`getInetAddressFromIntString`